### PR TITLE
32 adding a new item in wallet wont allow you to insert the first value

### LINKED ISSUE #32: Adding a new item in wallet won't allow you to insert the first value
Example: 
 You create a new wallet (e.g. ING bank)
 The value of the wallet is 0
 You have to insert all your income and expenses in order to set the correct value of the wallet

Be careful:
 When you add the possibility to insert the value of the wallet, this may already include some income/outcome on your expenses.   So, i'd be great to add the option to not insert the expense in the wallet.

### PR DESCRIPTION
**Pull request by simo2898**
The branch contains even some functionality for the new recurring billings

### CODE REVIEW

File: src/main/java/com/expensetracker/controller/ExpensePeriodController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+@RestController
+@RequestMapping("/api")
+@RequiredArgsConstructor
+public class ExpensePeriodController {
+
+    private final ExpensePeriodServiceImpl expensePeriodService;
+
+    @GetMapping("/period")
+    List<ExpensePeriod> getExpenses() {
```
Comment (mattiapitossi) [inline src/main/java/com/expensetracker/controller/ExpensePeriodController.java]: Maybe we should use ResponseEntity here

File: src/main/java/com/expensetracker/controller/ExpensePeriodController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        return expensePeriodService.getAllExpensesPeriod();
+    }
+
+    @PostMapping("/period")
+    ResponseEntity<ExpensePeriod> createExpensePeriod(@Valid @RequestBody ExpensePeriod expensePeriod) throws URISyntaxException {
+        System.out.println("EXPENSE PERIODDDDDD _____ :::::::" + expensePeriod);
```
Comment (mattiapitossi) [inline src/main/java/com/expensetracker/controller/ExpensePeriodController.java]: Do we have a logger?

File: src/main/java/com/expensetracker/model/File.java [hunk 3]
```diff
@@ -16,9 +16,9 @@ public class File {
     @Id
     @Column(name = "id")
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    public Integer id;
+    private Integer id;
```
Comment (mattiapitossi) [inline src/main/java/com/expensetracker/model/File.java:19]: Should we use a Long as ID?

File: src/main/java/com/expensetracker/model/Wallet.java [hunk 4]
```diff
@@ -23,16 +23,6 @@ public class Wallet {
     @Column(name = "name")
     private String name;
 
-    @Transient
-    @JsonSerialize
+    @Column(name = "value")
     private Double value;
```
Comment (mattiapitossi) [inline src/main/java/com/expensetracker/model/Wallet.java:27]: I wouldn't use a Double for a value in a wallet, better to use BigDecimal

File: src/main/java/com/expensetracker/service/WalletService.java [hunk 5]
```diff
@@ -38,8 +38,7 @@ public void deleteById(Integer walletId) {
     }
 
     public Double getValue(Wallet wallet) {
-        Double value = 0.00;
-
+        double value = wallet.getValue();
```
Comment (mattiapitossi) [inline src/main/java/com/expensetracker/service/WalletService.java:41]: As per the comment above, this should be changed to BigDecimal

File: src/main/java/com/expensetracker/service/impl/ExpensePeriodServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    private final WalletRepository walletRepository;
+
+    private final SubcategoryRepository subcategoryRepository;
+
+    @Override
+    public ExpensePeriod createNewExpensePeriod(ExpensePeriod expensePeriod) {
+        Category category = categoryRepository.findByName(expensePeriod.getCategory().getName());
+        expensePeriod.setCategory(category);
+
+        if (expensePeriod.getSubcategory().getName() == null) {
```
Comment (mattiapitossi) [inline src/main/java/com/expensetracker/service/impl/ExpensePeriodServiceImpl.java:32]: If the name could be null, I'd use an optional here

File: src/test/java/com/expensetracker/controller/WalletControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.ArrayList;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(value = WalletController.class)
+class WalletControllerTest {
```
Comment (mattiapitossi) [inline src/test/java/com/expensetracker/controller/WalletControllerTest.java:23]: Cool! tests!
Maybe we should also try an integration test for the wallet (could be another task)

**Review by mattiapitossi — CHANGES_REQUESTED** [review]

File: src/main/java/com/expensetracker/controller/ExpensePeriodController.java (hunk 1 above)
Comment (simo2898) [reply src/main/java/com/expensetracker/controller/ExpensePeriodController.java]: Changed to ResponseEntity in all the controllers.

File: src/main/java/com/expensetracker/controller/ExpensePeriodController.java (hunk 2 above)
Comment (simo2898) [reply src/main/java/com/expensetracker/controller/ExpensePeriodController.java]: At the moment we don't have a logger, but the sout has been removed.

File: src/main/java/com/expensetracker/model/File.java (hunk 3 above)
Comment (simo2898) [reply src/main/java/com/expensetracker/model/File.java:19]: Created an issue for this change

File: src/main/java/com/expensetracker/model/Wallet.java (hunk 4 above)
Comment (simo2898) [reply src/main/java/com/expensetracker/model/Wallet.java:27]: As per the comment above, an issue has been created.

File: src/main/java/com/expensetracker/service/WalletService.java (hunk 5 above)
Comment (simo2898) [reply src/main/java/com/expensetracker/service/WalletService.java:41]: As per the comment above, an issue has been created.

**Review by mattiapitossi — APPROVED** [review]

**Pull request merged**